### DISCUSSION TITLE
Objekt-Koordinaten bei der Abfrage ausgeben

### DISCUSSION THREAD

**Discussion by marko-jer** [e-netze/webgis-community#5] [Q&A] [answered]
Kann man in der Tabelle einer Abfrage die Objektkoordinaten ausgeben?
z.B.: 
- bei Polygonen einen Punkt innerhalb der Fläche?
- bei Linien einen Punkt auf der Linie
- bei Punkten den Punkt

als einzelne Werte: X Y

und wenn ja, wie konfiguriert man das im CMS?

**Comment by gis-eni** [comment, marked as answer]
Ist momentan nicht vorgesehen. Theoretisch könntest du das in den Daten abbilden, zB über einen View, der über T-SQL Funktionen den Punkt bestimmt und einem Feld ausgibt.

Generell kann so etwas aber auch einmal im WebGIS umgesetzt werden. Die "Koordinaten" liegen nach der Abfrage eigentlich vor (an der Position des Markers). Allerdings ist hier wieder die Frage, wie du die Koordinaten brauchst: Koordinatensystem, Formatierung, ...

Die Koordinaten nutzt ja beispielsweise auch der "Navigationsbutton", den man in der custom.js definieren kann (zB Absprung in Google Maps):

```javascript
webgis.usability.singleResultButtons.push(
    {
        name: 'Google Navigation',
        url: 'https://maps.google.com/maps?daddr={lat},{lon}&ll=',
        img: webgis.css.imgResource('google-maps-26.png', 'tools')
    });
``` 
Findige User haben den auch bei schon einmal benutzt, um die "google" Koordinaten für eine Punkt der Browser Url zu bekommen :D 

Was wäre denn der konkrete Anwendungsfall? Vielleicht gibt es ja eine bessere Lösung als die Koordinaten einfach nur anzuzeigen.

**Reply by marko-jer** [reply]
Danke für den Tipp mit den **webgis.usability.singleResultButton.push**.

Ich habe die Übergabe der Koordinaten auch über **webgis.custom.tools.add** versucht
Da gibt es das Problem, dass bei der Übergabe des "sketch2d" Polygons als wkt  der letzte Punkt (=1.Punkt)  und die doppelte Klammer fehlen.
[http://giswiki.org/wiki/Well_Known_Text](http://giswiki.org/wiki/Well_Known_Text)

**Comment by gis-eni** [comment]
Ok, das mit dem **webgis.custom.tools.add** POLYGON((....)) hab ich umgesetzt und ist dann ab 5.22.4504 behoben.
